dex/src/critbit.rs

```diff
@@ -724,29 +753,16 @@ impl Slab {
         self.remove_by_key(self.get(self.find_max()?)?.key()?)
     }
 
-    #[cfg(test)]
-    fn traverse(&self) -> Vec<&LeafNode> {
-        fn walk_rec<'a>(slab: &'a Slab, sub_root: NodeHandle, buf: &mut Vec<&'a LeafNode>) {
-            match slab.get(sub_root).unwrap().case().unwrap() {
-                NodeRef::Leaf(leaf) => {
-                    buf.push(leaf);
-                }
-                NodeRef::Inner(inner) => {
-                    walk_rec(slab, inner.children[0], buf);
-                    walk_rec(slab, inner.children[1], buf);
-                }
-            }
+    pub fn iter(&self, descending: bool) -> LeafNodeIterator<'_> {
+        let mut stack = vec![];
```

Comment:
i think it's worth initializing this one with a large enough capacity

Reply:
+1 the bump allocator in solana runtime makes dynamic vector resizing deceptively brutal

Reply:
`Vec::with_capacity(self.header().leaf_count as usize);`